src/Components/ModuleForm/ModuleForm.js

```diff
@@ -0,0 +1,310 @@
+import React, { Component } from "react";
+import "./ModuleForm.css";
+import { Button, ListGroup, Card } from "react-bootstrap";
+
+class ModuleForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      enabled: "1",
+      userAction: "Add",
+      moduleName: "",
+      moduleList: JSON.parse(localStorage.getItem("moduleList")) || [],
+      grade: "A",
+      error: "",
+    };
+  }
+
+  handleSubmit = (event) => {
+    const { userAction, moduleList, moduleName, grade } = this.state;
+
+    console.log("handleSubmit");
+
+    if (moduleName === "") {
+      this.setState({
+        error: "Module Name cannot be empty",
+      });
+      event.preventDefault();
+      return;
+    }
+    // Add logic for adding modules
+    if (userAction === "Add") {
+      const newModule = { moduleName, grade };
+      const searchTerm = moduleName;
+      const existingModuleList = moduleList.filter(
+        (module) => module.moduleName === searchTerm
+      );
+      if (existingModuleList.length > 0) {
+        console.log("Remove duplicated module");
+        this.setState({
+          error: "Duplicate Module Name",
+          moduleList: [...moduleList],
+        });
+        event.preventDefault();
+        return;
+      } else {
+        this.setState({ error: null });
+      }
+
+      console.log("Add Module");
+      this.setState(
+        {
+          moduleList: [...moduleList, newModule],
+          moduleName: "",
+          grade: this.state.grade,
+          userAction: "Add",
+        },
+        () => {
+          localStorage.setItem(
+            "moduleList",
+            JSON.stringify(this.state.moduleList)
+          );
+        }
+      );
+
+      // Add logic for updating modules
+    } else if (userAction === "Update") {
+      const newModule = { moduleName, grade };
+      const updatedModuleList = moduleList.map((module) => {
+        if (module.moduleName === newModule.moduleName) {
+          return newModule;
+        } else {
+          return module;
+        }
```

Comment:
```suggestion
        if (module.moduleName === newModule.moduleName) return newModule;
        return module;
```

If you return early, no need to else statement. By cutting away the curly braces and the else statement, it is easier to read and less to write